packages/block-editor/src/hooks/color.js

```diff
@@ -276,6 +277,47 @@ export function ColorEdit( props ) {
 	);
 }
 
+/**
+ * This adds inline styles for color palette colors.
+ * Ideally, this is not needed and themes should load their palettes on the editor.
+ *
+ * @param  {Function} BlockListBlock Original component
+ * @return {Function}                Wrapped component
+ */
+export const withColorPaletteStyles = createHigherOrderComponent(
+	( BlockListBlock ) => ( props ) => {
+		const { name, attributes } = props;
+		const { backgroundColor, textColor } = attributes;
+		const { colors } = useSelect( ( select ) => {
+			return select( 'core/block-editor' ).getSettings();
+		}, [] );
+
+		if ( ! hasColorSupport( name ) || ! backgroundColor || ! textColor ) {
+			return <BlockListBlock { ...props } />;
+		}
+		const extraStyles = {
+			color: textColor
+				? getColorObjectByAttributeValues( colors, textColor )?.color
+				: undefined,
+			backgroundColor: backgroundColor
+				? getColorObjectByAttributeValues( colors, backgroundColor )
+						?.color
+				: undefined,
+		};
+
+		let wrapperProps = props.wrapperProps;
+		wrapperProps = {
+			...props.wrapperProps,
+			style: {
+				...extraStyles,
+				...props.wrapperProps?.style,
```

Comment:
Here we assume props.wrapperProps may be undefined (props.wrapperProps?) but previously we spread it without any verification (...props.wrapperProps). Should we add a verification?

Reply:
spreading doesn't cause errors when the value is undefined.